source/fundamentals/crud/write-operations/embedded-arrays.txt

```diff
@@ -0,0 +1,188 @@
+===========================
+Update Arrays in a Document
+===========================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you will learn how to update arrays in a document.
+
+To update an array, you must do the following:
+
+- Specify the update you want to perform
+- Specify what array elements to apply your update to
+- Perform an update operation using these specifications
+
+Sample Document
+~~~~~~~~~~~~~~~
+
+The following sections feature examples that update this sample
+document: 
+
+.. code-block:: json
+
+    { "_id": 1, "color": "green", "qty": [8, 12, 18] }
+
+The examples on this page use the ``findOneAndUpdate()`` method of the
+``MongoCollection`` class to retrieve and update the document. Each
+example uses an instance of the ``FindOneAndUpdateOptions`` class to
+have MongoDB retrieve the document after the update occurs. For
+more information on the ``findOneAndUpdate()`` method, see our Compound
+Operations guide <TODO>. 
+
+Specifying an Update
+--------------------
+
+To specify an update, use the ``Updates`` builder. The ``Updates``
+builder provides static utility methods to construct update
+specifications. For more information on using the ``Updates`` builder with
+arrays, see our :ref:`guide on the Updates builder <array_updates>`.
+
+The following example performs these actions: 
+
+- Query for the sample document
+- Add "17" to the ``qty`` array using the ``push()`` method of the ``Updates`` builder
+
+.. literalinclude:: /includes/fundamentals/code-snippets/UpdateArray.java
+   :language: java
+   :dedent:
+   :start-after: begin pushElementsExample
+   :end-before: end pushElementsExample
+
+The example above updates the original document to the following state:
+
+.. code-block:: json
+   :copyable: false
+
+   { "_id": 1, "color": "green", "qty": [8, 12, 18, 17] }
+
+Specifying Array Elements
+-------------------------
+
+You can specify which array elements to update using a positional
+operator. Positional operators can specify the first, all or certain
+array elements to update. 
+
+To specify elements in an array with positional operators, use **dot
+notation**. Dot notation is a property access syntax for navigating JSON
+objects. 
+
+For additional information, see the server manual entry on
+:manual:`dot notation </core/document/#std-label-document-dot-notation>`.
+
+The First Array Element
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To update the first element that matches your query filter, use the
```

Comment:
```suggestion
To update the first array element that matches your query filter, use the
```

Nit: I think this can be made clearer by saying "array element". I personally am not sure what an "element" is (can be inferred from context but it isn't a term I am familiar with).